ci/ray_ci/automation/weekly_green_metric.py

```diff
@@ -0,0 +1,38 @@
+from typing import List
+
+import click
+import github
+
+from ci.ray_ci.utils import logger
+from ray_release.test_automation.state_machine import (
+    TestStateMachine,
+    WEEKLY_RELEASE_BLOCKER_TAG,
+)
+
+
+@click.command()
+@click.option(
+    "--production",
+    is_flag=True,
+    show_default=True,
+    default=False,
+    help=("Persist weekly green metric to S3"),
+)
+def main(production: bool) -> None:
+    github = TestStateMachine.get_github()
+    repo = TestStateMachine.get_ray_repo()
+
+    blockers = _get_release_blockers(github, repo)
+    logger.info(f"Found {blockers.totalCount} release blockers")
+
+
+def _get_release_blockers(
+    github: github.Github, repo: github.Repository.Repository
+) -> List[github.Issue.Issue]:
+    release_blocker_label = repo.get_label(WEEKLY_RELEASE_BLOCKER_TAG)
+
+    return github.get_user().get_issues(state="open", labels=[release_blocker_label])
```

Comment:
can we move this into `state_machine` and have some unit tests?

Reply:
totally